shell/platform/embedder/embedder.cc

```diff
@@ -269,6 +325,47 @@ InferOpenGLPlatformViewCreationCallback(
     }
   };
 
+  auto gl_fbo_with_damage_callback =
+      [fbo_with_damage_callback = config->open_gl.fbo_with_damage_callback,
+       user_data](intptr_t id) -> flutter::GLFBOInfo {
+    // If no fbo_with_damage_callback was provided, disable partial repaint.
+    if (!fbo_with_damage_callback) {
+      return flutter::GLFBOInfo{
+          .fbo_id = static_cast<uint32_t>(id),
+          .partial_repaint_enabled = false,
+          .existing_damage = SkIRect::MakeEmpty(),
+      };
+    }
+
+    // Given the FBO's ID, get its existing damage.
+    FlutterDamage existing_damage;
+    fbo_with_damage_callback(user_data, id, &existing_damage);
```

Comment:
Is the expectation that the damage rect buffers that the embedder sends us can be collection when the present callback for the same `fbo_id` is called? If so, let us document this in the header.

Reply:
Yes, that is the expectation. I will update the header to be more specific about this behavior.